### PR TITLE
Operator metrics

### PR DESCRIPTION
**Pull request by lgajowy**
Closes #559 
___

## Summary:

Reinstates operator, demand, and processed messages metrics. Refactors the code a bit

### CODE REVIEW

**Comment by lgajowy** [comment]
There's probably lots of code that could be refactored together with that pr but it would take too much time for one blow. So I decided to create this PR and any other cleanups/refactorings will be done in later phases (0.9.0 for example)

File: otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/akka/stream/AkkaStreamMonitorExtension.scala [hunk 1]
```diff
@@ -8,64 +8,183 @@ import akka.actor.typed.ExtensionId
...
+      currentSnapshot.groupBy(_._2.subStreamName.streamName.name)
 
-    val streamShells = current.groupBy { case (ref, _) =>
-      stream.subStreamNameFromActorRef(ref)
-    }
-    val streamNames = streamShells.keySet.map(_.streamName)
-    metrics.setRunningStreamsTotal(streamNames.size)
-    metrics.setRunningActorsTotal(current.size)
+    val stageSnapshots: Seq[StageSnapshot] = currentSnapshot.values.flatMap { streamInfo =>
+      val subStreamName = streamInfo.subStreamName
```
Comment (ptrdom) [inline otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/akka/stream/AkkaStreamMonitorExtension.scala:70]: Unused variable.

File: core/src/main/scala/io/scalac/mesmer/core/model/stream/ConnectionsIndexCache.scala [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      }.asScala
+
+    new ConnectionsIndexCache(mutableMap)
+  }
+
+  /**
+   * Exists solely for testing purpose
+   * @return
+   */
+  def empty = new ConnectionsIndexCache(mutable.Map.empty)
```
Comment (ptrdom) [inline core/src/main/scala/io/scalac/mesmer/core/model/stream/ConnectionsIndexCache.scala:104]: Probably should not exist in main sources if it is only needed for testing.

**Review by ptrdom — CHANGES_REQUESTED** [review]
Small nitpicks only. I cannot do more than a baseline code review - for me, this functionality just has too much code with too little context to be comprehended comfortably by a person that is not the author.

**Review by ptrdom — APPROVED** [review]

**Pull request merged**